sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -126,18 +143,41 @@ case class ConcatWs(children: Seq[Expression])
       // All children are strings. In that case we can construct a fixed size array.
       val evals = children.map(_.genCode(ctx))
 
-      val inputs = evals.map { eval =>
-        s"${eval.isNull} ? (UTF8String) null : ${eval.value}"
-      }.mkString(", ")
-
-      ev.copy(evals.map(_.code).mkString("\n") + s"""
-        UTF8String ${ev.value} = UTF8String.concatWs($inputs);
+      val argNums = evals.length
+      val args = ctx.freshName("args")
+
+      val inputs = evals.tail.zipWithIndex.map { case (eval, index) =>
+        if (eval.isNull != "true") {
+          s"""
+           ${eval.code}
```

Comment:
nit: in this and the next 4 lines an indentation space is missing if I am not wrong.

Reply:
good catch